smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -187,4 +211,48 @@ public void closeQuietly() {
             log.exceptionOnClose(e);
         }
     }
+
+    public void isAlive(HealthReport.HealthReportBuilder builder) {
+        if (configuration.getHealthEnabled()) {
+            List<Throwable> actualFailures;
+            boolean isCurrentlyConnected;
+            synchronized (this) {
+                actualFailures = new ArrayList<>(failures);
+            }
+            if (!actualFailures.isEmpty()) {
+                builder.add(configuration.getChannel(), false,
+                        actualFailures.stream().map(Throwable::getMessage).collect(Collectors.joining()));
+            } else {
+                builder.add(configuration.getChannel(), true);
+            }
+        }
+
+        // If health is disable do not add anything to the builder.
+    }
+
+    private String getTopic() {
+        return configuration.getTopic().orElseGet(configuration::getChannel);
```

Comment:
As this is now being used in at least a couple of places, should we be capturing the topic name instead of calling configuration more than once?

Reply:
Yes, good idea! Just did it.